### ISSUE TITLE
About masked acceleration

### ISSUE THREAD

**Issue by Harry-Miral** [thu-ml/SageAttention#127]
Can I still use SageAttention to speed up if I manually provide att_mask (not causal nor padding mask)？Thank you.

**Comment by jt-zhang** [comment]
Hi, I think `SpargeAttn` may have done the thing you mentioned: https://github.com/thu-ml/SpargeAttn.

**Comment by Harry-Miral** [comment]
Oh my goodness.
from sageattention import sageattn
import torch.nn.functional as F

F.scaled_dot_product_attention = sageattn
So I just use the above part and get amazing speedup with my custom mask?

**Comment by jt-zhang** [comment]
Hi, Have you read the SpargeAttn code? Please note that it is different from SageAttention.

**Comment by Harry-Miral** [comment]
Also, does it support different length data within the same batch?

**Comment by jason-huang03** [comment]
@Harry-Miral Currently not. What kind of workload are you working on? If the batch size is quite small, then you can use for loop on each single batch of data.
